pkg/webhook/webhook.go

```diff
@@ -63,6 +63,7 @@ var (
 	injectHugepageDownApi  bool
 	resourceNameKeys       []string
 	honorExistingResources bool
+	customizedInjections   = make(map[string]string)
```

Comment:
could you be specific to the unexpected result? my understanding is if there is a wrong format in customized injection , or incomplete patch or unsupported patch exists in customized Injection, it would cause the createCustomizedPatch fail, the end result would be no customized patch be applied to pod.

Reply:
> could you be specific to the unexpected result?

`SetCustomizedInjection()` first adds new entries then removes stale entires.
As an example for what might go wrong, a user defines a new config replacing all entries, the map is then updated with new configs added.
Then a pod is created, leading to the injections of both new and old configs and only then the "remove stale entries" part in `SetCustomizedInjection()` takes place.

In general you have a data structure that is written and read asynchronously in a non atomic manner by different goroutines, it is generally best practice to add some sort of synchronization mechanism to avoid any inconsistencies when reading the data structure.

Reply:
Added a read/write lock for global structure data update.